Sync/BookmarkPayload.swift

```diff
@@ -61,35 +61,33 @@ public enum BookmarkType: String {
     }
 
     public static func payloadFromJSON(json: JSON) -> BookmarkBasePayload? {
+        if json["deleted"].asBool ?? false {
+            // Deleted records won't have a type.
+            return BookmarkBasePayload(json)
+        }
+
         guard let typeString = json["type"].asString else {
             return nil
         }
+
         guard let type = BookmarkType.init(rawValue: typeString) else {
```

Comment:
nit: Is there a reason why these are 2 separate guard statements and not all wrapped in one, seeing as they both just return nil? I guess there is a little extra clarity here, but it feels a tad redundant.

Also, I know this was not added in this patch, but when you notice something....